README.rst

```diff
@@ -320,6 +328,32 @@ current user; and ``"OS_ROOT"`` is used to open the journal from directories rel
 directory path or file descriptor. Multiple flags can be OR'ed together using a list:
 ``["LOCAL_ONLY", "CURRENT_USER"]``.
 
+``secret_filters`` (default ``[]``)
+
+Secret filters can be used to redact sensitive data which matches known patterns in logs before forwarding the message along
+to it's final destination. To use: add a number of filters following the pattern below to the reader config. The ``pattern`` is a standard
+python regex with groups. ``redact_groups`` defines which groups contents will be replaced, with the ``redaction_message``. Patterns are
+compiled at runtime.
+
+
+"secret_filters": [
+  {
+    "pattern": "(.*?\\s?)(SECRET)(\\s?.*)",
+    "redact_groups": [1],
+    "redaction_message": "[REDACTED]"
```

Comment:
Not sure if this was discussed somewhere but was there some particular reason for going with this sort of syntax? For simple cases this seems pretty complex and also error prone as your lines will just be truncated if you forgot the (.*) after the string you're actually looking for. Pattern to look for and allowing referencing match groups in redaction message for more advanced use cases might be easier to use. I.e. the above would be expressed with

```
{
    "pattern": "SECRET",
    "replacement": "[REDACTED]"
}
```

Then you could also do something like

```
{
    "pattern": "(some prefix) SECRET (followed by)",
    "replacement": "\1 [REDACTED] \2"
}
```

To do "This is some prefix SECRET followed by something" => "This is some prefix [REDACTED] followed by something". Also allows more complex transformations as needed and implementation is just `re.sub(pattern, replacement, log_line)`

Reply:
this was a great suggestion, implemented now. the concept of redaction groups is gone, and both formats of regex will work / have examples / tests